### ISSUE TITLE
Enhancement: [naming-convention] Add an optional identifier member of each option

### ISSUE THREAD

**Issue by yf-yang** [typescript-eslint/typescript-eslint#9129]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/naming-convention

### Description

This feature is used to help debug naming-convention options.

It is pretty hard to debug naming-convention, as it is hard to determine which option is actually activated.

Currently, to ease the pain, the order is documented here in details: https://typescript-eslint.io/rules/naming-convention/#how-does-the-rule-automatically-order-selectors
Additionally, the error message would show which selector is matched, and what those valid cases are.

However, they are still many other properties that determines the order, many options could share the same format and selector..., so it is still hard to figure out which option is actually working.

The proposal is: adding an optional string property, called `name` or `id` or whatever. If that property is provided, then the identifier would be reported to the user along with other information.

### Fail

```TypeScript
// noop
```

### Pass

```TypeScript
// noop
```
